### PR TITLE
[kotlin2cpg] Implemented Lambda Parameter Destruction

### PR DESCRIPTION
**Pull request by max-leuthaeuser**
This will most likely change/break sptests. 
@mpollmeier I was not able to figure out how to get buildbot running with this PR. Could you check? (Maybe we should add a short description in CS).

### CODE REVIEW

File: joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/ast/AstForFunctionsCreator.scala [hunk 1]
```diff
@@ -186,9 +188,93 @@ trait AstForFunctionsCreator(implicit withSchemaValidation: ValidationMode) {
     )
   }
 
+  private def astsForDestructing(param: KtParameter): Seq[Ast] = {
```
Comment (DavidBakerEffendi) [inline joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/ast/AstForFunctionsCreator.scala:191]: A nitpick
```suggestion
  private def astsForDestructuring(param: KtParameter): Seq[Ast] = {
```

File: joern-cli/frontends/kotlin2cpg/src/test/scala/io/joern/kotlin2cpg/dataflow/LambdaTests.scala [hunk 2]
```diff
@@ -20,20 +21,15 @@ class LambdaTests extends KotlinCode2CpgFixture(withOssDataflow = true) {
...
-//          List(
-//            ("f1(p)", Some(3)),
-//            ("p to 1", Some(4)),
-//            ("mapOf(p to 1, \"two\" to 2, \"three\" to 3)", Some(4)),
-//            ("val m = mapOf(p to 1, \"two\" to 2, \"three\" to 3)", Some(4)),
-//            ("m.forEach { (k, v) ->  println(k) }", Some(5)),
-//            ("<lambda>0(k, v)", Some(5)),
-//            ("println(k)", Some(5))
-//          )
+          List(
```
Comment (DavidBakerEffendi) [inline joern-cli/frontends/kotlin2cpg/src/test/scala/io/joern/kotlin2cpg/dataflow/LambdaTests.scala:26]: Thanks for getting this back up!

**Review by DavidBakerEffendi — APPROVED** [review]

**Comment by mpollmeier** [comment]
buildbot is happy with this pr

**Comment by max-leuthaeuser** [comment]
Thanks!

**Pull request merged**